Odata-docs/webapi-8/routing-overview.md

```diff
@@ -0,0 +1,365 @@
+
+# [.NET 6.0](#tab/net60)
+
+```csharp
+// Program.cs
+var builder = WebApplication.CreateBuilder(args);
+
+var modelBuilder = new ODataConventionModelBuilder();
+modelBuilder.EntitySet<Customer>("Customers");
+
+builder.Services.AddControllers().AddOData(
+    options => options.EnableQueryFeatures(maxTopValue: null).AddRouteComponents(
+        routePrefix: "odata",
+        model: modelBuilder.GetEdmModel()));
+
+var app = builder.Build();
+```
+
+# [.NET Core 3.1](#tab/netcoreapp31)
+
+```csharp
+// Startup.cs
+public void ConfigureServices(IServiceCollection services)
+{
+    var modelBuilder = new ODataConventionModelBuilder();
+    modelBuilder.EntitySet<Customer>("Customers");
+
+    services.AddControllers().AddOData(
+        options => options.EnableQueryFeatures(maxTopValue: null).AddRouteComponents(
+            routePrefix: "odata",
+            model: modelBuilder.GetEdmModel()));
+}
+```
+
+--
+
+## OData routing
+In the above block of code, we build an Edm model using the `ODataConventionModelBuilder`. We then call the `AddOData()` method resulting into the following OData routing conventions being registered in the [dependency injection](/aspnet/core/fundamentals/dependency-injection) container:
+- `MetadataRoutingConvention` - routes requests for service metadata
+- [`EntitySetRoutingConvention`](/odata/webapi-8/entityset-routing) - routes requests for entity sets
+- [`EntityRoutingConvention`](/odata/webapi-8/entity-routing) - routes requests for single entities by key
+- [`SingletonRoutingConvention`](/odata/webapi-8/singleton-routing) - routes requests for singletons
+- [`FunctionRoutingConvention`](/odata/webapi-8/function-routing) - routes requests for Edm functions
+- [`ActionRoutingConvention`](/odata/webapi-8/action-routing) - routes requests for Edm actions
+- `OperationImportRoutingConvention` - routes requests for operation imports
+- `PropertyRoutingConvention` - routes requests for structural properties
+- [`NavigationRoutingConvention`](/odata/webapi-8/navigation-routing) - routes requests for navigation properties
+- [`RefRoutingConvention`](/odata/webapi-8/ref-routing) - routes requests for entity reference links
+- `AttributeRoutingConvention` - represents a routing convention for attribute-routed requests
+
+A in-depth walkthrough for the core routing conventions can be found in the relevant tutorial.
+
+At application startup, OData path templates for all potential endpoints are built. The OData routing conventions above are "executed" to generate a route table that maps each controller action (endpoint) to an OData route template.
```

Comment:
all conventions get executed at startup?

Reply:
I noticed you didn't mention some of the conventions: MetadataRoutingConvention, OperationImportRoutingConvention and PropertyRoutingConvention

Reply:
@ElizabethOkerio "executed", "run", "triggered", etc. Basically what we do is try to apply each convention to the controller and controller actions defined in the project with the output of that exercise being the route table that we render when someone visits the `/$odata` page. This is unlike 7.x where we used to try and find a match upon receiving the request. In 8.x it's just a matter of looking up from the pre-generated route table. I don't know if there's a better word that I can use in place of "executed". It's actually the reason I placed it between quotes

Reply:
@ElizabethOkerio I'll do a separate doc for `PropertyRoutingConvention`. There are far too many route templates covered by that convention and I realized the rest would get seriously delayed. There other two are trivial: `MetadataRoutingConvention` is used to route the `/$metadata` requests. But I'll make sure I cover them in some sort of way